prediction-applier/src/integration-test/java/org/alfresco/hxi_connector/prediction_applier/domain/usecase/e2e/PredictionApplicationIntegrationTest.java

```diff
@@ -25,30 +25,57 @@
  */
 package org.alfresco.hxi_connector.prediction_applier.domain.usecase.e2e;
 
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Test;
+import org.springframework.test.annotation.DirtiesContext;
 
 import org.alfresco.hxi_connector.prediction_applier.domain.usecase.e2e.util.PredictionApplierE2ETestBase;
 
 @SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
```

Comment:
Isn't there any way to avoid this? E.g.: manually resetting some dirtied properties / using beforeAll | afterAll appropriately? I'm not entirely aware of the reason why we added this, but it can be quite expensive as more methods get added to this class.

Reply:
At the moment - there isn't. User agent data is lazy loaded and is not being refreshed ever since as application is running (maybe we could do that from time to time, but still, I treat it as kind of static data)
If the test method which executes a happy path is run before the second method, then application context already has repository version pulled from ACS and will not fail, however we'd like it to fail on that in prediction applier.